ui/tests/acceptance/secrets/backend/database/secret-test.js

```diff
@@ -407,37 +437,38 @@ module('Acceptance | secrets/database/*', function (hooks) {
   });
 
   test('buttons show up for managing connection', async function (assert) {
-    const backend = await mount();
+    const backend = this.backend;
     const connection = await newConnection(backend);
-    await connectionPage.visitShow({ backend, id: connection });
-    assert
-      .dom('[data-test-database-connection-delete]')
-      .hasText('Delete connection', 'Delete connection button exists with correct text');
-    assert
-      .dom('[data-test-database-connection-reset]')
-      .hasText('Reset connection', 'Reset button exists with correct text');
-    assert.dom('[data-test-secret-create]').hasText('Add role', 'Add role button exists with correct text');
-    assert.dom('[data-test-edit-link]').hasText('Edit configuration', 'Edit button exists with correct text');
     const CONNECTION_VIEW_ONLY = `
-      path "${backend}/*" {
-        capabilities = ["deny"]
-      }
       path "${backend}/config" {
         capabilities = ["list"]
       }
       path "${backend}/config/*" {
         capabilities = ["read"]
       }
+      # allow backend cleanup on afterEach
+      path "sys/mounts/${backend}" {
+        capabilities = ["delete"]
+      }
     `;
-    await consoleComponent.runCommands([
-      `write sys/mounts/${backend} type=database`,
-      `write sys/policies/acl/test-policy policy=${btoa(CONNECTION_VIEW_ONLY)}`,
-      'write -field=client_token auth/token/create policies=test-policy ttl=1h',
+    const token = await runCmd(consoleComponent, [
+      ...createPolicyCmd('test-policy', CONNECTION_VIEW_ONLY),
+      ...tokenWithPolicyCmd('test-policy'),
     ]);
-    const token = consoleComponent.lastTextOutput;
-    await logout.visit();
+    await navToConnection(backend, connection);
```

Comment:
Visiting the connection page directly was not re-querying the capabilities correctly. I was unable to replicate this behavior in the app, so navigating via the UI seems to resolve the test issue